### PR TITLE
Refactor Person object to Entry Object, modified commands to work for new Entry Object

### PR DESCRIPTION
**Pull request by JasonYapzx**
## Status

COMPLETE

## Description

Added the classes for the `Entry` class to keep track of expenditures/income. The class is currently comprised of the following:
* Description: used as an identifier
* Amount: amount spent
* Date: date of the expense
* Tags: (included for future feature implementation)

## Checklist
- [x] Create Entry class
- [x] Change entries to accept `Entry` objects instead of just `Person` objects from AB3
- [x] Change command to delete `Entry` entries in the application
- [x] Saving functionality for `Entry` objects work

## Screenshots
NIL

### CODE REVIEW

File: src/main/java/seedu/address/model/entry/Date.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+public class Date {
+    public static final String MESSAGE_CONSTRAINTS =
+            "Dates should be of the format dd-mm-yyyy and it should only contain numbers";
+    public static final String VALIDATION_REGEX = "^([0-2][0-9]||3[0-1])-(0[1-9]||[1-9]||1[0-2])-([0-9][0-9])?[0-9][0-9]$";
+    ;;
+    public final String date;
```
Comment (yuanxi1) [inline src/main/java/seedu/address/model/entry/Date.java:11]: Should we use `LocalDate` instead? Right now if I put 31-02-2022 it correctly adds the entry and does not return any error but it's an invalid date.

Comment (JasonYapzx) [reply src/main/java/seedu/address/model/entry/Date.java:11]: Oh yea.. my bad the validation regex only checks for the validity of a string, can we change it after the base content has been merged?

File: src/main/java/seedu/address/model/expenditure/Description.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        return other == this // short circuit if same object
+                || (other instanceof Description // instanceof handles nulls
+                && fullDescription.equals(((Description) other).fullDescription)); // state check
+    }
+
+    @Override
+    public int hashCode() {
+        return fullDescription.hashCode();
+    }
+}
```
Comment (EmilyOng) [inline src/main/java/seedu/address/model/expenditure/Description.java:57]:
```suggestion
}

```

I think there's 2 `Description.java` without newline. The other one was fixed

Comment (JasonYapzx) [reply src/main/java/seedu/address/model/expenditure/Description.java:57]: Oh yes.. I forgot I have not deleted the other `Description.java` file, thanks!

**Pull request merged**